lib/api/apiUtils/object/checkHttpHeadersSize.js

```diff
@@ -0,0 +1,25 @@
+const { errors } = require('arsenal');
+const { maxHttpHeadersSize } = require('../../../../constants');
+
+/**
+ * Checks the size of the size of the HTTP headers
+ * @param {object} requestHeaders - HTTP request headers
+ * @return {object} object with error or null
+ */
+function checkHttpHeadersSize(requestHeaders) {
+    let httpHeadersSize = 0;
+
+    Object.keys(requestHeaders).forEach(header => {
+        httpHeadersSize += Buffer.byteLength(header, 'utf8') +
```

Comment:
Just want to clarify this simply takes the number of bytes for the header and value, but does not count any separators or quotes we would need if we were to send the headers as a request. Not sure if that makes any sense...

Reply:
I am not sure I understand the issue being raised. Wouldn't quotes/separators be included as part of the header values (`;` or any escaped quotes)?

Reply:
(https://scality.atlassian.net/browse/S3C-5072?focusedCommentId=293869) describes the rationale for the change

Reply:
So in the request you see the headers as:
`url?h1=val1&h2=val2` etc
Just want to make sure the "header size" is not the total size of that string and is instead just the key and values. That is, the equals and ampersand are ignored.

Reply:
I don't think the querystring values would factor in the header size count